test/api_compatibility.js

```diff
@@ -104,30 +87,66 @@ module.exports = function apiCompatibilityChecks(_createTracer) {
         });
 
         describe('SpanContext', function() {
-            var tracer = createTracer();
-            var span = tracer.startSpan('test_span');
-            var spanContext = span.context();
-
             it('should be a class', function() {
+                var spanContext = span.context();
                 expect(spanContext).to.be.an('object');
             });
         });
 
-
         describe('Reference', function() {
-            var tracer = createTracer();
-            var span = tracer.startSpan('test_span');
-            var ref = new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span.context());
 
             it('should be a class', function() {
+                var ref = new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span.context());
                 expect(ref).to.be.an('object');
             });
 
             it('should handle Spans and SpanContexts', function() {
-                var tracer = new opentracing.Tracer();
-                var span = tracer.startSpan('test_operation');
                 expect(function() { new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span); }).to.not.throw(Error);
             });
         });
     });
+
+    describe('Opentracing implementation compatibility', function() {
+        var tracer;
+        var span;
+
+        beforeEach(function() {
+            tracer = createTracer();
+            span = tracer.startSpan();
+        });
+
+        describe('Tracer', function() {
+            it ('startSpan returns span with allowed functions', function() {
+                assert.ok(hasSpanFunctions(span));
+            });
+
+            it ('extract returns span context with allowed functions', function() {
+                // pass by default if no factory map was specified.
+                if (carrierFormatFactoryMap) {
+                    var carrierFactory = carrierFormatFactoryMap[opentracing.FORMAT_TEXT_MAP];
+                    var spanContext = tracer.extract(opentracing.FORMAT_TEXT_MAP, carrierFactory());
+                    assert.isOk(hasSpanContextFunctions(spanContext));
+                }
+            });
+        });
+
+        describe('Span', function() {
+            it('context#should return a span context', function() {
+                assert.isOk(hasSpanContextFunctions(span.context()));
+            });
+
+            it('tracer#should return a tracer', function() {
+                assert.isOk(hasTracerFunctions(span.tracer()));
+            });
+
+            it('should set baggage and retrieve baggage', function() {
+                span.setBaggageItem('some-key', 'some-value');
+                var val = span.getBaggageItem('some-key');
+                assert.isOk(val);
+                if (checkBaggageValues) {
+                    assert.equal('some-value', val);
+                }
+            });
+        });
```

Comment:
what about all other span methods like setTag, log, etc. Even if we cannot validate the return value (sometimes we can, setTag are chainable), we at least can check that those methods are callable without exceptions.